2000 New Jersey Congressional Districts

KT0416838 opened the pull request:
## Redistricting requirements
In ``New Jersey``, according to [NCSL Redistricting Law 2000](https://web.archive.org/web/20041216185957/https://www.senate.mn/departments/scr/redist/red2000/Tab5appx.htm), districts must:

1. be geographically contiguous
2. have equal populations

### Algorithmic Constraints
We enforce a maximum population deviation of 0.5%. We use a pseudo-county constraint described below, which attempts to mimic the norms in ``New Jersey`` of generally preserving county and municipal boundaries.
We add thresholded hard constraints to ensure at least one black performant district.

## Data Sources
Data for ``New Jersey`` comes from the [ALARM Project's update](https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/ZV5KF3) to [The Record of American Democracy](https://road.hmdc.harvard.edu/).

## Pre-processing Notes
Because ``New Jersey`` contains an unusually large number of discontiguous precincts, we used a helper function to identify 20 discontiguity merge-groups, involving 46 precincts (0.75% of all VTDs). These units were merged prior to simulation.
We apply a county-level logit shift to recalibrate ``ndv/nrv``. The adjustment needed to correct the vote shares was fairly large in several counties: Essex required a logit shift of about 1.39, Hudson about 1.33, Passaic about 0.98, and Union about 1.07. Because some of these values exceed 1, the original search interval [-1, 1] was too narrow for ``uniroot()`` to find the solution, so we widened the search interval to [-1.5, 1.5] to accommodate the larger correction.

## Simulation Notes
We sample 5,000 districting plans for ``New Jersey`` across 5 independent runs of the SMC algorithm.
To improve mixing, we apply the merge-split procedure at every step of the algorithm and allow up to 20 updates within each step.
To balance county and municipality splits, we create pseudo-counties for use in the county constraint. 

## Validation
<img width="3000" height="4500" alt="validation_20260315_1743" src="https://github.com/user-attachments/assets/72acf9cf-1f5e-4b5a-898d-cf0eedb2b3e8" />



```
SMC with Merge-Split MCMC Steps: 5,000 sampled plans of 13 districts on 6,138 units
Plans sampled on Spanning Forest Space using the Uniform Valid Edge forward kernel.
SMC Parameters:
• `weight_type` = optimal
• `seq_alpha` = 1
• `pop_temper` = 0.01
Mergesplit Parameters:
• `total_ms_steps` = 12
• `mh_accept_per_smc` = 20
• `pair_rule` = uniform
Plan diversity 80% range: 0.40 to 0.62
Largest R-hat values for ordered summary statistics:
    comp_edge   comp_polsby county_splits   muni_splits       ndshare           ndv           nrv      plan_dev      pop_aian 
        1.002         1.003         1.001         1.001         1.011         1.012         1.011         1.001         1.016 
    pop_asian     pop_black      pop_hisp      pop_nhpi     pop_other   pop_overlap       pop_two     pop_white     total_vap 
        1.009         1.013         1.019         1.012         1.009         1.016         1.011         1.019         1.006 
     vap_aian     vap_asian     vap_black      vap_hisp      vap_nhpi     vap_other       vap_two     vap_white 
        1.011         1.013         1.015         1.014         1.015         1.008         1.016         1.018 
• R-hat ≤ 1.05: 338
• 1.05 < R-hat ≤ 1.1: 0
• R-hat > 1.1: 0
• Total R-hats: 338
Sampling diagnostics for SMC run 1 of 5 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique 
Split 1       198 (19.8%)     99.5%        1.15   645 (102%) 
Split 2       310 (31.0%)     78.4%        1.02   416 ( 66%) 
Split 3       427 (42.7%)     61.1%        0.86   456 ( 72%) 
Split 4       462 (46.2%)     53.5%        0.73   485 ( 77%) 
Split 5       637 (63.7%)     42.2%        0.65   509 ( 81%) 
Split 6       685 (68.5%)     35.1%        0.59   543 ( 86%) 
Split 7       711 (71.1%)     31.0%        0.55   540 ( 85%) 
Split 8       785 (78.5%)     25.9%        0.48   560 ( 89%) 
Split 9       815 (81.5%)     22.3%        0.45   554 ( 88%) 
Split 10      867 (86.7%)     18.9%        0.39   573 ( 91%) 
Split 11      890 (89.0%)     16.1%        0.34   570 ( 90%) 
Split 12      934 (93.4%)     13.5%        0.28   528 ( 84%) 
Resample      934 (93.4%)       NA%          NA   895 (142%) 

Sampling diagnostics for Mergesplit Steps of SMC run 1 of 5 (5,000 samples)
           Acc. rate MS Steps per Plan
MS Step 1      41.8%                40
MS Step 2      32.5%                60
MS Step 3      27.7%                80
MS Step 4      25.3%                80
MS Step 5      23.8%                80
MS Step 6      22.4%               100
MS Step 7      22.7%               100
MS Step 8      22.7%               100
MS Step 9      23.1%               100
MS Step 10     23.4%               100
MS Step 11     24.6%               100
MS Step 12     25.4%               100

•  Watch out for low effective samples, very low acceptance rates (less than 1%), large std. devs. of the log weights (more than 3 or
so), and low numbers of unique plans. R-hat values for summary statistics should be between 1 and 1.05.
```

## Additional Notes
BVAP performance plot:
<img width="893" height="686" alt="ecent Black by VAR" src="https://github.com/user-attachments/assets/25b85d52-f624-46d3-91ea-49f62557cce9" />


Total Black performant districts:
```
  n_black_perf    n
1            1 4986
2            2   14
```

## Checklist

- [x] I have followed the [instructions](https://github.com/alarm-redist/fifty-states/blob/main/CONTRIBUTING.md)
- [x] I have updated the [tracker](https://docs.google.com/spreadsheets/d/1k_tYLoE49W_DCK1tcWbouoYZFI9WD76oayEt5TOmJg4/edit#gid=453387933)
- [x] All `TODO` lines from the template code have been removed
- [x] I have merged in the main branch and then recalculated summary statistics
- [x] I have run `enforce_style()` to format my code
- [x] The documentation copied above is up-to-date 
- [x] There are no data files in this pull request
- [x] None of the file output paths (for the `redist_map` and `redist_plans` objects, and summary statistics) have been edited

@christopherkenny

KT0416838 committed:
Add files via upload

KT0416838 committed:
Update 02_setup_NJ_cd_2000.R

christopherkenny commented on analyses/NJ_cd_2000/03_sim_NJ_cd_2000.R:
```diff
@@ -0,0 +1,34 @@
+
+set.seed(2000)
+plans <- redist_smc(map, nsims = 4e3, runs = 10, counties = pseudo_county)
+# IF CORES OR OTHER UNITS HAVE BEEN MERGED:
```
Can we remove these lines?

KT0416838 replied:
Yes, I removed it!

KT0416838 committed:
Update 03_sim_NJ_cd_2000.R

KT0416838 committed:
Update doc_NJ_cd_2000.md

christopherkenny commented:
For example, the Japan project did: https://github.com/alarm-redist/japan/blob/main/function/check_valid.R

tylersimko commented on analyses/NJ_cd_2000/03_sim_NJ_cd_2000.R:
```diff
@@ -0,0 +1,32 @@
+
+cli_process_done()
+cli_process_start("Saving {.cls redist_plans} object")
+
```
@KT0416838 We're on a meeting right now, but we could add the contiguity check that we are discussing with @christopherkenny here.

kosukeimai commented:
> For example, the Japan project did: https://github.com/alarm-redist/japan/blob/main/function/check_valid.R

Thanks for reminding me @christopherkenny Yeah, now I remember that in the Japan project we had lots of discontiguous cases.

KT0416838 committed:
Update 02_setup_NJ_cd_2000.R

KT0416838 committed:
Update doc_NJ_cd_2000.md

KT0416838 committed:
Update 03_sim_NJ_cd_2000.R

KT0416838 commented:
Thank you, @kosukeimai @tylersimko @christopherkenny . I’ve updated the code to include the function, and it now produces only contiguous plans. The R-hat values are also below 1.05. 
@tylersimko I’ve updated the PR description as well. Could you please review them?

<img width="3000" height="4500" alt="validation_20251018_0614" src="https://github.com/user-attachments/assets/97b95382-027c-4f3a-814d-06acc295e94d" />

```
SMC: 5,000 sampled plans of 13 districts on 6,138 units
Plans sampled on Graph Space using the Naive Top K forward kernel.
Forward kernel parameters: `adapt_k_thresh`=0.99
SMC Parameters:
• `weight_type` = optimal
• `seq_alpha` = 1
• `pop_temper` = 0
Plan diversity 80% range: 0.42 to 0.65
Largest R-hat values for ordered summary statistics:
    comp_edge   comp_polsby county_splits   muni_splits       ndshare 
        1.004         1.012         1.007         1.007         1.022 
          ndv           nrv      plan_dev      pop_aian     pop_asian 
        1.015         1.014         1.007         1.013         1.015 
    pop_black      pop_hisp      pop_nhpi     pop_other   pop_overlap 
        1.020         1.018         1.016         1.014         1.004 
      pop_two     pop_white     total_vap      vap_aian     vap_asian 
        1.021         1.017         1.016         1.016         1.015 
    vap_black      vap_hisp      vap_nhpi     vap_other       vap_two 
        1.018         1.019         1.010         1.013         1.020 
    vap_white 
        1.013 
• R-hat ≤ 1.05: 338
• 1.05 < R-hat ≤ 1.1: 0
• R-hat > 1.1: 0
• Total R-hats: 338
Sampling diagnostics for SMC run 1 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1     4,901 (27.2%)     16.7%        1.03 11,383 (100%)     10 
Split 2     6,074 (33.7%)     26.4%        0.87  7,694 ( 68%)      6 
Split 3     7,183 (39.9%)     34.7%        0.78  8,669 ( 76%)      4 
Split 4     7,259 (40.3%)     32.5%        0.74  9,120 ( 80%)      4 
Split 5     8,710 (48.4%)     29.9%        0.68  9,336 ( 82%)      4 
Split 6     9,918 (55.1%)     26.9%        0.61  9,705 ( 85%)      4 
Split 7     5,867 (32.6%)     24.2%        0.63  9,961 ( 88%)      4 
Split 8     3,753 (20.9%)     15.3%        0.70  9,540 ( 84%)      6 
Split 9    10,770 (59.8%)     19.0%        0.62  9,124 ( 80%)      4 
Split 10    9,335 (51.9%)     15.4%        0.52  9,901 ( 87%)      4 
Split 11    7,922 (44.0%)     15.7%        0.66  9,344 ( 82%)      3 
Split 12   17,124 (95.1%)      6.4%        0.23  8,225 ( 72%)      3 
Resample   17,124 (95.1%)       NA%          NA 16,423 (144%)     NA 

Sampling diagnostics for SMC run 2 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1     4,953 (27.5%)     13.8%        1.01 11,159 ( 98%)     12 
Split 2     6,024 (33.5%)     23.1%        0.87  7,812 ( 69%)      7 
Split 3     7,247 (40.3%)     21.7%        0.78  8,701 ( 76%)      7 
Split 4     7,655 (42.5%)     32.7%        0.73  9,139 ( 80%)      4 
Split 5     7,622 (42.3%)     36.7%        0.69  9,442 ( 83%)      3 
Split 6     8,478 (47.1%)     26.7%        0.63  9,597 ( 84%)      4 
Split 7     5,518 (30.7%)     24.0%        0.65  9,859 ( 87%)      4 
Split 8     3,470 (19.3%)     15.4%        0.72  9,485 ( 83%)      6 
Split 9    10,610 (58.9%)     15.4%        0.62  9,041 ( 79%)      5 
Split 10    6,685 (37.1%)     15.4%        0.54  9,950 ( 87%)      4 
Split 11    9,514 (52.9%)     15.5%        0.65  9,354 ( 82%)      3 
Split 12   17,167 (95.4%)      8.7%        0.23  8,438 ( 74%)      2 
Resample   17,167 (95.4%)       NA%          NA 16,434 (144%)     NA 

Sampling diagnostics for SMC run 3 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1     4,906 (27.3%)     16.5%        1.03 11,201 ( 98%)     10 
Split 2     5,549 (30.8%)     26.7%        0.89  7,706 ( 68%)      6 
Split 3     7,256 (40.3%)     29.1%        0.79  8,598 ( 76%)      5 
Split 4     7,149 (39.7%)     20.6%        0.74  9,046 ( 80%)      7 
Split 5     8,013 (44.5%)     25.5%        0.69  9,305 ( 82%)      5 
Split 6     8,189 (45.5%)     27.4%        0.62  9,557 ( 84%)      4 
Split 7     5,147 (28.6%)     29.7%        0.65  9,845 ( 87%)      3 
Split 8     4,006 (22.3%)     15.3%        0.70  9,445 ( 83%)      6 
Split 9    10,419 (57.9%)     19.1%        0.61  9,187 ( 81%)      4 
Split 10    8,159 (45.3%)     19.6%        0.53  9,935 ( 87%)      3 
Split 11    9,166 (50.9%)     20.9%        0.66  9,341 ( 82%)      2 
Split 12   17,107 (95.0%)      8.6%        0.24  8,494 ( 75%)      2 
Resample   17,107 (95.0%)       NA%          NA 16,291 (143%)     NA 

Sampling diagnostics for SMC run 4 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1     4,949 (27.5%)     15.0%        1.01 11,148 ( 98%)     11 
Split 2     6,018 (33.4%)     23.0%        0.87  7,858 ( 69%)      7 
Split 3     7,167 (39.8%)     29.4%        0.78  8,776 ( 77%)      5 
Split 4     7,363 (40.9%)     17.8%        0.74  9,052 ( 80%)      8 
Split 5     8,088 (44.9%)     25.1%        0.69  9,418 ( 83%)      5 
Split 6     9,438 (52.4%)     32.6%        0.62  9,612 ( 84%)      3 
Split 7     5,139 (28.6%)     24.2%        0.64  9,954 ( 87%)      4 
Split 8     4,193 (23.3%)     27.2%        0.69  9,417 ( 83%)      3 
Split 9    11,101 (61.7%)     30.2%        0.61  9,172 ( 81%)      2 
Split 10    9,684 (53.8%)     24.6%        0.53  9,967 ( 88%)      2 
Split 11    8,184 (45.5%)     20.9%        0.68  9,321 ( 82%)      2 
Split 12   17,070 (94.8%)      9.3%        0.24  8,109 ( 71%)      2 
Resample   17,070 (94.8%)       NA%          NA 16,368 (144%)     NA 

Sampling diagnostics for SMC run 5 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1     4,966 (27.6%)     16.6%        1.02 11,240 ( 99%)     10 
Split 2     6,028 (33.5%)     26.5%        0.88  7,775 ( 68%)      6 
Split 3     6,878 (38.2%)     25.0%        0.78  8,691 ( 76%)      6 
Split 4     7,629 (42.4%)     27.4%        0.74  9,077 ( 80%)      5 
Split 5     7,885 (43.8%)     18.5%        0.69  9,379 ( 82%)      7 
Split 6     8,625 (47.9%)     16.3%        0.62  9,625 ( 85%)      7 
Split 7     6,290 (34.9%)     20.0%        0.63  9,872 ( 87%)      5 
Split 8     4,137 (23.0%)     27.2%        0.70  9,600 ( 84%)      3 
Split 9    10,852 (60.3%)     29.7%        0.62  9,175 ( 81%)      2 
Split 10    8,184 (45.5%)     19.1%        0.53  9,926 ( 87%)      3 
Split 11    8,221 (45.7%)      8.5%        0.66  9,290 ( 82%)      6 
Split 12   17,150 (95.3%)      4.8%        0.23  8,365 ( 74%)      4 
Resample   17,150 (95.3%)       NA%          NA 16,421 (144%)     NA 

Sampling diagnostics for SMC run 6 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1     4,947 (27.5%)     18.4%        1.02 11,179 ( 98%)      9 
Split 2     5,695 (31.6%)     26.7%        0.88  7,743 ( 68%)      6 
Split 3     7,952 (44.2%)     34.7%        0.78  8,641 ( 76%)      4 
Split 4     6,633 (36.9%)     27.4%        0.75  9,182 ( 81%)      5 
Split 5     8,198 (45.5%)     30.1%        0.69  9,221 ( 81%)      4 
Split 6     7,687 (42.7%)     16.4%        0.63  9,686 ( 85%)      7 
Split 7     5,736 (31.9%)     19.9%        0.63  9,780 ( 86%)      5 
Split 8     3,310 (18.4%)     15.4%        0.70  9,558 ( 84%)      6 
Split 9    10,682 (59.3%)     18.9%        0.61  9,077 ( 80%)      4 
Split 10    8,340 (46.3%)     15.2%        0.54  9,863 ( 87%)      4 
Split 11    8,382 (46.6%)     16.1%        0.67  9,305 ( 82%)      3 
Split 12   17,162 (95.3%)      8.8%        0.23  8,522 ( 75%)      2 
Resample   17,162 (95.3%)       NA%          NA 16,421 (144%)     NA 

Sampling diagnostics for SMC run 7 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1     5,089 (28.3%)     16.7%        1.02 11,237 ( 99%)     10 
Split 2     5,895 (32.8%)     23.1%        0.88  7,789 ( 68%)      7 
Split 3     7,479 (41.5%)     34.4%        0.77  8,670 ( 76%)      4 
Split 4     7,446 (41.4%)     33.0%        0.73  9,211 ( 81%)      4 
Split 5     8,134 (45.2%)     25.1%        0.68  9,280 ( 82%)      5 
Split 6     8,025 (44.6%)     16.1%        0.62  9,694 ( 85%)      7 
Split 7     7,038 (39.1%)     16.7%        0.63  9,912 ( 87%)      6 
Split 8     3,935 (21.9%)     21.6%        0.69  9,604 ( 84%)      4 
Split 9    10,810 (60.1%)     15.7%        0.62  9,236 ( 81%)      5 
Split 10    8,361 (46.4%)     15.6%        0.54  9,926 ( 87%)      4 
Split 11    8,585 (47.7%)     16.0%        0.68  9,172 ( 81%)      3 
Split 12   17,209 (95.6%)      9.1%        0.22  8,350 ( 73%)      2 
Resample   17,209 (95.6%)       NA%          NA 16,372 (144%)     NA 

Sampling diagnostics for SMC run 8 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1     4,878 (27.1%)     20.8%        1.03 11,164 ( 98%)      8 
Split 2     5,585 (31.0%)     26.7%        0.89  7,793 ( 68%)      6 
Split 3     6,829 (37.9%)     34.9%        0.78  8,621 ( 76%)      4 
Split 4     7,649 (42.5%)     39.6%        0.74  9,137 ( 80%)      3 
Split 5     8,143 (45.2%)     21.0%        0.69  9,370 ( 82%)      6 
Split 6     7,847 (43.6%)     19.0%        0.62  9,703 ( 85%)      6 
Split 7     5,465 (30.4%)     24.2%        0.66  9,845 ( 87%)      4 
Split 8     3,322 (18.5%)     27.4%        0.73  9,315 ( 82%)      3 
Split 9    10,941 (60.8%)     29.3%        0.62  9,088 ( 80%)      2 
Split 10    9,053 (50.3%)     24.3%        0.54  9,951 ( 87%)      2 
Split 11    8,361 (46.5%)     20.7%        0.68  9,109 ( 80%)      2 
Split 12   17,192 (95.5%)      6.9%        0.23  7,990 ( 70%)      3 
Resample   17,192 (95.5%)       NA%          NA 16,476 (145%)     NA 

Sampling diagnostics for SMC run 9 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1     4,940 (27.4%)     15.2%        1.03 11,249 ( 99%)     11 
Split 2     6,018 (33.4%)     23.0%        0.87  7,774 ( 68%)      7 
Split 3     7,664 (42.6%)     25.5%        0.78  8,648 ( 76%)      6 
Split 4     6,887 (38.3%)     33.1%        0.75  9,272 ( 81%)      4 
Split 5     8,063 (44.8%)     30.0%        0.68  9,327 ( 82%)      4 
Split 6     8,581 (47.7%)     19.2%        0.63  9,606 ( 84%)      6 
Split 7     5,912 (32.8%)     23.9%        0.63  9,849 ( 87%)      4 
Split 8     3,383 (18.8%)     27.1%        0.70  9,485 ( 83%)      3 
Split 9    11,020 (61.2%)     15.6%        0.61  9,207 ( 81%)      5 
Split 10    9,664 (53.7%)     19.5%        0.52  9,929 ( 87%)      3 
Split 11    8,104 (45.0%)     21.0%        0.66  9,464 ( 83%)      2 
Split 12   17,208 (95.6%)      8.7%        0.23  8,455 ( 74%)      2 
Resample   17,208 (95.6%)       NA%          NA 16,446 (145%)     NA 

Sampling diagnostics for SMC run 10 of 10 (5,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1     4,876 (27.1%)     18.4%        1.02 11,229 ( 99%)      9 
Split 2     5,961 (33.1%)     17.9%        0.87  7,708 ( 68%)      9 
Split 3     6,957 (38.6%)     29.2%        0.78  8,725 ( 77%)      5 
Split 4     7,712 (42.8%)     20.6%        0.73  9,169 ( 81%)      7 
Split 5     7,493 (41.6%)     25.3%        0.69  9,387 ( 83%)      5 
Split 6     8,556 (47.5%)     26.9%        0.63  9,622 ( 85%)      4 
Split 7     4,878 (27.1%)     12.8%        0.65  9,761 ( 86%)      8 
Split 8     4,139 (23.0%)     18.4%        0.71  9,421 ( 83%)      5 
Split 9    11,130 (61.8%)     18.7%        0.60  9,205 ( 81%)      4 
Split 10    9,831 (54.6%)     15.2%        0.54  9,796 ( 86%)      4 
Split 11    9,231 (51.3%)     15.4%        0.64  9,231 ( 81%)      3 
Split 12   17,130 (95.2%)      4.7%        0.23  8,326 ( 73%)      4 
Resample   17,130 (95.2%)       NA%          NA 16,391 (144%)     NA 

•  Watch out for low effective samples, very low acceptance rates (less than
1%), large std. devs. of the log weights (more than 3 or so), and low numbers
of unique plans. R-hat values for summary statistics should be between 1 and
1.05.
```

KT0416838 committed:
Update 02_setup_NJ_cd_2000.R

KT0416838 committed:
Update doc_NJ_cd_2000.md

tylersimko reviewed (changes requested):
Thanks @KT0416838! I agree, this looks good. 

One small request: could you please provide evidence that the contiguity check is working in the PR? The example plans look good, but ideally we could have a table attached here that runs @christopherkenny's contiguity function on the final simulated plans to confirm 100% of them are fully contiguous.

KT0416838 committed:
Update doc_NJ_cd_2000.md

KT0416838 committed:
Update 02_setup_NJ_cd_2000.R

KT0416838 committed:
Update 03_sim_NJ_cd_2000.R

KT0416838 committed:
Update 03_sim_NJ_cd_2000.R

KT0416838 committed:
Update 03_sim_NJ_cd_2000.R

KT0416838 committed:
Update 03_sim_NJ_cd_2000.R

KT0416838 committed:
Update 02_setup_NJ_cd_2000.R

KT0416838 committed:
Update 01_prep_NJ_cd_2000.R

KT0416838 committed:
Update 02_setup_NJ_cd_2000.R

KT0416838 committed:
Update 03_sim_NJ_cd_2000.R

KT0416838 committed:
Update doc_NJ_cd_2000.md

KT0416838 committed:
Update doc_NJ_cd_2000.md

KT0416838 committed:
Update doc_NJ_cd_2000.md

christopherkenny reviewed (changes requested):
Just a few small things re VRA and some code cleanups.

KT0416838 commented:
@christopherkenny 
Regarding the partially missing election returns, I've also been examining the cause and suspect this is due to a subdivision-code mismatch during the MCD-to-VTD allocation step in the construction of `NJ_2000.csv`.
I compared `nj_2000$mcds` with `nj_shp$mcd`, and all VTDs with `ndv = nrv = 0` have `mcd` values that are not present in the `mcds` universe.
Could you share the code used to construct `NJ_2000.csv`, particularly the MCD/MCDGRP-to-VTD mapping step? If this diagnosis is correct, we likely need to harmonize the subdivision-code mapping (e.g., by adjusting the merge key used during CSV construction).

KT0416838 committed:
Update doc_NJ_cd_2000.md

KT0416838 committed:
Update 01_prep_NJ_cd_2000.R

KT0416838 committed:
Update 02_setup_NJ_cd_2000.R

KT0416838 committed:
Update 03_sim_NJ_cd_2000.R

KT0416838 commented:
@christopherkenny 
I updated this PR and code based on our previous discussions.

1. Regarding the missing election returns data in Essex, the baseline vote data appears too Republican (Democratic shares far below the 2000 MEDSL targets) in several core counties (e.g., Essex ~0.44 vs. 0.76; Hudson ~0.43 vs. 0.74; Passaic ~0.35 vs. 0.59; Union ~0.34 vs. 0.60), likely reflecting the earlier bad merge caused by broken MCDGRP identifiers in Essex, as also suggested by the pre-shift simulation plot attached below. This distortion made hinge and rejection constraints difficult to satisfy. I therefore applied the logit shift. When applying it, the required adjustment lay outside the original `uniroot()` interval `[-1,1]`, so I widened the interval to `[-5,5]`.
2. I imposed a rejection-sampling constraint to ensure that each plan has at least one black opportunity district.
3. I added the number of merged discontiguous precincts to the documentation.

<img width="714" height="571" alt="Screenshot 2026-03-13 at 0 47 23" src="https://github.com/user-attachments/assets/66c08ae8-ad89-4d3c-8dd4-dc2646b95d7a" />

After these adjustments, all the plans contain at least one black opportunity district, as shown in the plot in the PR description. Every update is reflected in the PR description and the code. Please feel free to review them.

christopherkenny commented on analyses/NJ_cd_2000/doc_NJ_cd_2000.md:
```diff
@@ -0,0 +1,24 @@
+## Pre-processing Notes
+We use a helper function to identify 20 discontiguity merge-groups, involving 46 precincts (0.75% of all VTDs), which were merged prior to simulation.
+Because the Essex MCDGRP identifiers are broken and produce a distorted baseline merge, several counties have raw Democratic shares far below the 2000 MEDSL targets. 
+We apply a county-level logit shift to recalibrate ``ndv/nrv`` and widen the ``uniroot()`` search interval from ``[-1,1]`` to ``[-5,5]`` to accommodate the larger correction.
```
Can you elaborate on this?

christopherkenny commented on analyses/NJ_cd_2000/doc_NJ_cd_2000.md:
```diff
@@ -0,0 +1,24 @@
+
+## Simulation Notes
+We sample 5,000 districting plans for ``New Jersey`` across 5 independent runs of the SMC algorithm.
+We also use the algorithmic mergesplit parameters to improve mixing.
```
We should specify the number of steps.

KT0416838 committed:
Update 01_prep_NJ_cd_2000.R

KT0416838 committed:
Update 02_setup_NJ_cd_2000.R

KT0416838 committed:
Update 03_sim_NJ_cd_2000.R

KT0416838 committed:
Update doc_NJ_cd_2000.md

KT0416838 committed:
Update doc_NJ_cd_2000.md

KT0416838 committed:
Update doc_NJ_cd_2000.md

KT0416838 committed:
Update 01_prep_NJ_cd_2000.R

KT0416838 committed:
Update 01_prep_NJ_cd_2000.R

christopherkenny reviewed (approved):
Thanks, Kento.
